cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/TopicConfigProvider.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.config;
+
+import com.linkedin.cruisecontrol.common.CruiseControlConfigurable;
+import java.util.Map;
+import java.util.Properties;
+
+
+public interface TopicConfigProvider extends CruiseControlConfigurable, AutoCloseable {
+
+  /**
+   * Get cluster-level configs that applies to a topic if no topic-level config exist for it.
+   */
+  Properties clusterConfigs();
+
+  /**
+   * Get topic-level configs for the requested topic.
+   * @param topic Topic for which the topic-level configs are requested.
+   */
+  Properties topicConfigs(String topic);
```

Comment:
* It may not yet be used in our doe, but having this interface provides us more options.
* `KafkaTopicConfigProvider` is a particular implementation of `TopicConfigProvider`. Depending on the goal, it can be customized to involve caching, or another source to get this information.
Also, using this method would be considered as more efficient if user needs only the configs for a single topic, but not all topics. This way, the processing at `ZooKeeper` side, and the response size will be smaller -- i.e. less `CPU` and `outbound network` load.